v2/moviebuff.go

```diff
@@ -0,0 +1,336 @@
+//
+// resourceType may be one of the following value defined by ResourceType
+// page and limit is optional. Provide zero values to ignore them.
+//
+// limit represents the number of records to fetch in a single request.
+// The actual count can be lower than the provided limit. Max value is 50.
+// page represents the page number in the pagination. It starts from 1.
+func (m *moviebuff) GetResources(ctx context.Context, resourceType ResourceType, limit, page int) (*Resources, error) {
+	u := "/resources/" + string(resourceType) + "?"
+	if limit != 0 {
+		u += "limit=" + strconv.Itoa(limit)
+	}
+
+	if page != 0 {
+		u += "page=" + strconv.Itoa(page)
+	}
+
+	r, err := prepareRequest(ctx, m.HostURL, m.StaticToken, u)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := m.Client.Do(r)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		if res.StatusCode == http.StatusForbidden {
+			return nil, ErrInvalidToken
+		}
+		if res.StatusCode == http.StatusNotFound {
+			return nil, ErrResourceDoesNotExist
+		}
+		return nil, ErrResponseNotReceived
+	}
+
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := new(Resources)
+	err = json.Unmarshal(content, resources)
+	if err != nil {
+		return nil, err
+	}
+
+	return resources, nil
+}
+
+// GetCertifications fetches a list of all certifications available on Moviebuff
+
+//GetCertifications takes an optional argument country which can be the Qube Wire Cinemas country UUID or the ISO 2-digit code for this country, eg "IN". If country is provided, GetCertifications returns certifications available for the given country
+//Pass empty value for country to get a list of all certifications across countries
+func (m *moviebuff) GetCertifications(ctx context.Context, country string) ([]Certification, error) {
+	r, err := prepareRequest(ctx, m.HostURL, m.StaticToken, "/certifications")
+	if err != nil {
+		return nil, err
+	}
+	if country != "" {
+		addQueryParams(r, map[string]string{"country": country})
+	}
+
+	res, err := m.Client.Do(r)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		if res.StatusCode == http.StatusForbidden {
+			return nil, ErrInvalidToken
+		}
+		if res.StatusCode == http.StatusNotFound {
+			return nil, ErrResourceDoesNotExist
+		}
+		return nil, ErrResponseNotReceived
+	}
+
+	content, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	certifications := struct {
+		Data []Certification `json:"data"`
+	}{}
+	err = json.Unmarshal(content, &certifications)
+	if err != nil {
+		return nil, err
+	}
+
+	return certifications.Data, nil
+}
+
+func (m *moviebuff) GetHolidayCalendar(ctx context.Context, countryID string) (*Calendar, error) {
+	r, err := prepareRequest(ctx, m.HostURL, m.StaticToken, "/holidays/"+countryID)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := m.Client.Do(r)
```

Comment:
Feel like we should be able to wrap v1 here rather than rewriting it again. Let's discuss in person so we are in line with approach @thamizhv